### PR TITLE
Add interfaces and class for basics

### PR DESCRIPTION
**Pull request by omuomugin**
Added some basic class and interfaces that is defined in https://github.com/cryptoeconomicslab/WakkanayWallet/tree/master/src/ovm .

Implementation of them is not included, if there are something missing or something redundant just let me know.

### CODE REVIEW

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.cryptoeconomicslab.furano_common.types
+
+typealias Bytes = String
```
Comment (omuomugin) [inline furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt]: [Q]
Do we need this kind of typealias?

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/db/KeyValueStore.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.cryptoeconomicslab.furano_common.db
+
+import com.cryptoeconomicslab.furano_common.types.Bytes
+import kotlinx.coroutines.Deferred
```
Comment (omuomugin) [inline furano-common/src/main/java/com/cryptoeconomicslab/furano_common/db/KeyValueStore.kt]: [Memo]
`Deferred` is something similar to Promise or Future but it is in Kotlin Coroutine (which is light weighted thread and use to represent asynchronous programming)

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/db/RangeStore.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.cryptoeconomicslab.furano_common.db
+
+import com.cryptoeconomicslab.furano_common.types.Bytes
+import kotlinx.coroutines.Deferred
+
+interface RangeStore {
+    fun getAsync(start: Long, end: Long): Deferred<Bytes>
```
[the remark this answers is not in the document]
Comment (syuhei176) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/db/RangeStore.kt]: It's 32bytes. But I think it's ok at this time.

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.cryptoeconomicslab.furano_common.types
+
+data class Decision(
+    val outcome: Boolean
```
Comment (syuhei176) [inline furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt]: We needed `implicationProofs` in the previous spec.
But in the next spec,  we need a game tree.

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt (hunk 1 above)
Comment (omuomugin) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt]: In this context, is `byte` means 8-bit signed integer ? or like arrays of Char.
There is actually `Byte` type and `ByteArray` type in Kotlin and that represents 8-bit signed integer and list of 8-bit signed integers.

So if we want String, String will be fitted type.

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt (hunk 4 above)
Comment (omuomugin) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt]: Ok you mean this, https://github.com/cryptoeconomicslab/WakkanayWallet/blob/master/src/ovm/types/types.ts#L11 I missed that.

Should I just add `implicationProofs` now and after the spec was fixed we can change it to game tree maybe?

Comment (syuhei176) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt]: yes!

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt (hunk 1 above)
Comment (omuomugin) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt]: @syuhei176 Reminding this

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt (hunk 4 above)
Comment (omuomugin) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Decision.kt]: fixed in https://github.com/cryptoeconomicslab/Furano/pull/2/commits/ea5d0a505e4e7b918ca15d0c7dac95c41d2c0883

File: furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt (hunk 1 above)
Comment (syuhei176) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt]: I think `ByteArray` or more concrete class is better. for example we are using `Bytes` in Rust

Comment (omuomugin) [reply furano-common/src/main/java/com/cryptoeconomicslab/furano_common/types/Bytes.kt]: Ok I'll be changing to `Byte` and `ByteArray`

**Comment by omuomugin** [comment]
I fixed some of the issue on conversations so I'll merge those

**Pull request merged**